### PR TITLE
Fix gas support

### LINKED ISSUE #1767: Investigate differences in gas support between `quaireaux` and `protostar`
They seem to be using [this](https://github.com/keep-starknet-strange/quaireaux/blob/main/quaireaux/utils/src/utils.cairo):
```cairo
#[inline(always)]
fn check_gas() {
    match gas::withdraw_gas_all(get_builtin_costs()) {
        Option::Some(_) => {},
        Option::None(_) => {
            let mut data = ArrayTrait::new();
            data.append('Out of gas');
            panic(data);
        }
    }
}
```

Meanwhile in our tests we use [this](https://github.com/software-mansion/protostar/blob/master/tests/integration/cairo1/gas_test/test.cairo#L5):
```cairo
    match gas::withdraw_gas() {
        Option::Some(_) => {},
        Option::None(_) => {
            let mut data = ArrayTrait::new();
            data.append('Out of gas');
            panic(data);
        }
    }
```

`withdraw_gas_all(get_builtin_costs())` causes [this]() e2e test using quaireaux as dependency fail.

### PR DESCRIPTION
**Pull request by cptartur**
Closes #1767 

### Checklist
- [x] Relevant issue is linked
- [x] Docs updated/issue for docs created
- [x] Added relevant tests

### CODE REVIEW

File: tests/integration/cairo1/misc/gas_test/test.cairo [hunk 1]
```diff
@@ -2,7 +2,7 @@ use array::ArrayTrait;
 
 
 fn fib(a: felt252, b: felt252, n: felt252) -> felt252 {
-    match gas::withdraw_gas() {
+    match gas::withdraw_gas_all(get_builtin_costs()) {
```
Comment (karol-bisztyga) [inline tests/integration/cairo1/misc/gas_test/test.cairo:5]: could you please explain this change a bit?

**Review by karol-bisztyga — COMMENTED** [review]
bless

File: tests/integration/cairo1/misc/gas_test/test.cairo (hunk 1 above)
Comment (cptartur) [reply tests/integration/cairo1/misc/gas_test/test.cairo:5]: This shouldn't change anything in the behavior (I think/hope), but as per #1767 description, the syntax I introduced didn't work previously.

Perhaps we can have separate tests for both `withdraw_gas` and `withdraw_gas_all`?

Comment (karol-bisztyga) [reply tests/integration/cairo1/misc/gas_test/test.cairo:5]: > Perhaps we can have separate tests for both `withdraw_gas` and `withdraw_gas_all`? 

That wouldn't be a bad idea.

**Review by karol-bisztyga — APPROVED** [review]
I think it is ok, please address that one open comment from me, thank you.

File: tests/integration/cairo1/misc/gas_test/test.cairo (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply tests/integration/cairo1/misc/gas_test/test.cairo:5]: > Perhaps we can have separate tests for both withdraw_gas and withdraw_gas_all?
I would include them here

File: protostar/cairo/cairo_function_runner_facade.py [hunk 2]
```diff
@@ -53,8 +53,9 @@ def run_from_offset(
             )
 
     # pylint: disable=unused-argument
-    @staticmethod
+    # @staticmethod
```
Comment (MaksymilianDemitraszek) [inline protostar/cairo/cairo_function_runner_facade.py:56]: .

**Review by MaksymilianDemitraszek — APPROVED** [review]

File: tests/integration/cairo1/misc/gas_test/test.cairo (hunk 1 above)
Comment (cptartur) [reply tests/integration/cairo1/misc/gas_test/test.cairo:5]: Updated tests

Comment (piotmag769) [reply tests/integration/cairo1/misc/gas_test/test.cairo:5]: My description was probably a bit unclear, but I didn't mean that using `withdraw_gas` in the tests made using quaireaux fail. It was rather an observation - `withdraw_gas` worked with protostar, but quaireaux used `withdraw_gas_all(get_builtin_costs())` which made protostar fail.

**Comment by MaksymilianDemitraszek** [comment]
@cptartur Can we fix and merge this one?

**Pull request merged**